### ISSUE TITLE
How to configure HTTPS access with Docker [question] 

### ISSUE THREAD

**Issue by cts1085** [zwave-js/zwavejs2mqtt#748]
I have attempted a few different ways to get the openssl running in a docker container so that accessing the UI from HA (which currently is accessed via HTTPS) works...  I am using a YML file with docker-compose and I just can't seem to get the right combination of environment variables and cert locations.  

I have the certs located on PI in /ssl and i have tried mapping that to /ssl and /etc/ssl/private with no luck.

When I set HTTPS='true' it seems to keep restarting the container.

is there any guidance on turning on this feature?

My current YML with options commented out is:

```
version: '3.7'
services:
  zwavejs2mqtt:
    container_name: zwavejs2mqtt
#    image: zwavejs/zwavejs2mqtt:latest
    image: zwavejs/zwavejs2mqtt:master
    restart: unless-stopped
    tty: true
    stop_signal: SIGINT
    hostname: HHHHH
    domainname: DDDDD.com
    environment:
      - TZ=America/New_York
#      - HTTPS='true'
#      - HOST=HHHHH.DDDDD.com
      - PORT=8091
#      - SSL_KEY=/ssl/privkey.pem
#      - SSL_CERTIFICATE=/ssl/cert.pem
    networks:
      - zwave
    devices:
      - '/dev/ttyACM0:/dev/ttyACM0'
    volumes:
      - ./store:/usr/src/app/store
      - /ssl:/etc/ssl/private
    ports:
      - '8091:8091' # port for web interface
      - '3000:3000' # port for zwave-js websocket server
networks:
  zwave:
# volumes:
#   zwavejs2mqtt:
#     name: zwavejs2mqtt

```

**Comment by robertsLando** [comment]
LOGs please

**Comment by cts1085** [comment]
the log is very short - the container seems to restart after just a few seconds.  I let it run for a minute to get multiple cycles.
The only change I made from the previous yml was to uncomment - HTTPS='true'

[zwavejs2mqtt-store-https.zip](https://github.com/zwave-js/zwavejs2mqtt/files/6049656/zwavejs2mqtt-store-https.zip)

**Comment by robertsLando** [comment]
@cts1085 The keys are automatically genereted and stored on `store` folder. By checking your logs I can only see it stops when it tries to generete the keys but I don't see the error causing the restart

**Comment by cts1085** [comment]
So I should not be placing any cert files in the store folder? I will remove those and see if that helps.

so I commented out hostname/domainname, removed the storage reference to /ssl and it seems to start now.

However, the certificate is just for localhost and accessing it from other devices/locations does not seem to work. Is there a way I can use my own certs that include the hostname of the PI box?

In my network I am running HASS from a Ubuntu instance and running Zwave2MQTT or ZwaveJS2MQTT in a docker container on a raspberry pi.

When I try to use https to access ZwaveJS2MQTT directly it tells me the certificate is invalid.

So... I copied my certs into the store folder using the same names and permissions and it WORKS!!!

So the trick is to make sure that my certs are named the same as what is being expected.

Thank you!

Now that it is working from a browser, i need to see how to get it to launch correctly from within HA.  On to that challenge next!
